### ISSUE TITLE
[TypeScript] The `id` attribute is optinal.

### ISSUE THREAD

**Issue by jooohn** [hashicorp/terraform-cdk#214]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Most AWS resources have `id` attribute. In my understanding, the `id` value should not be undefined, but actually its type is `string | undefined`. Generated codes look like the following.

```typescript
  // id - computed: true, optional: true, required: false
  private _id?: string;
  public get id() {
    return this._id ?? this.getStringAttribute('id');
  }
  public set id(value: string | undefined) {
    this._id = value;
  }
```

`get id()` itself can return `string` rather than `string | undefined`. However we have `set id()` function which receives `string | undefined`, and there is a restriction that the get / set accessor must have the same type (https://github.com/microsoft/TypeScript/issues/2521). Simply replacing `set id()` with regular method like `setId()` solves this issue.

Is there something I missed, or do you have any idea about this issue?

### References

https://github.com/microsoft/TypeScript/issues/2521

**Comment by skorfmann** [comment]
Yes, I noticed this as well and that's related to #25 / #26. 

There's also #188 which will likely include reworking the accessor logic a bit and most likely will address this as well.

**Comment by skorfmann** [comment]
Fixed by #310
